climpred_skeleton/comparison.py

```diff
@@ -0,0 +1,71 @@
+from typing import Union
+
+import pkg_resources
+import xarray as xr
+import yaml
+
+from .core import Verification
+
+
+class Comparison(Verification):
+    """First step in pipeline. Broadcast your prediction and observations
+    in a certain manner.
+
+    Base Comparison class hints at which attributes are expected for each
+    sub comparison class. It also stores the factory for internal use, to
+    return the appropriate subfunction.
+    """
+
+    _hindcast_comparison: bool
+    _probabilistic: bool
+
+    def __init__(self, prediction: xr.Dataset, observation: xr.Dataset) -> None:
+        super().__init__(prediction, observation)
+
+    def factory(self, type) -> 'Comparison':
+        return comparison_dict[type](self._prediction, self._observation)
+
+
+class e2o(Comparison):
+    """Example subclass for Comparison.
+
+    Stores useful information (e.g. whether this is a probabilistic comparison)
+    and houses main `broadcast` function, returning the modified prediction
+    and observation objects.
+    """
+
+    def __init__(self, prediction: xr.Dataset, observation: xr.Dataset) -> None:
+        super().__init__(prediction, observation)
+        self._hindcast_comparison = True
+        self._probabilistic = False
+
+    def broadcast(self) -> Union[xr.Dataset, xr.Dataset]:
+        if 'member' in self._prediction.dims:
+            prediction = self._prediction.mean('member')
+        else:
+            prediction = self._prediction
+        return prediction, self._observation
+
+
+class m2o(Comparison):
+    def __init__(self, prediction: xr.Dataset, observation: xr.Dataset) -> None:
+        super().__init__(prediction, observation)
+        self._hindcast_comparison = True
+        self._probabilistic = True
+
+    def broadcast(self) -> Union[xr.Dataset, xr.Dataset]:
+        observation = self._observation.expand_dims({'member': self._nmember})
+        observation['member'] = self._members
+        return self._prediction, observation
+
+
+comparison_yaml_file = pkg_resources.resource_filename(
```

Comment:
Has to go at bottom to avoid circular imports..

Reply:
What's the circular import issue? We should get rid of circular imports ASAP before things get more complicated.